### PR TITLE
[Tablet Orders] Create order: Address design review items

### LINKED ISSUE #12155: [Tablet Orders] Create Order: UI updates
From 6kkCwI9VOEYcidp6UT5bln-fi-1108_27535

* ✅ Barcode button doesn’t do anything: Expected for this iteration, has been removed.
* ✅ Reduce gap size
* [x] TODO: Missing divider lines
* [x] TODO: 
  * ✅56pt high buttons. 
  * ✅ No gap between buttons. 
  * ✅ 8pt space above and below button groups - even groups of one (top and bottom of section).

Screenshot 2024-02-29 at 19 58 39

### PR DESCRIPTION
**Pull request by iamgabrielma**
Closes: #12155

## Description
This PR tackles the design review items logged for "Create Order", that is:

Screenshot 2024-03-01 at 14 23 23

* Adds missing divider lines wrapping the `AddOrderComponentsSection` ("+ Add Shipping", "+ Add Coupon", ... )
* Makes all tappable buttons/rows 56pt height
* Reduces the existing space above and below button groups from 16pt to 8pt
* Adds a 8pt space to the top of the view

I have applied these without a conditional  `sideBySideViewForOrderForm` feature flag usage, since are design updates that are not coming only with this project but should affect the Order Form view across all its instances (split view, modal on modal, or plain iPhone). Happy to add additional logic if you think we should keep the previous UI if the flag is off.

## Testing instructions
1. With the feature flag off, observe that changes are applied to modal-on-modal, and iPhone
simulator_screenshot_446BA2A5-074C-47BA-8476-A1A5ED79DC30

2. With the feature flag on, observe that changes are applied to side-by-side, and iPhone
20240301-changes-applied-side-by-side

Adding products, shipping, etc, ... should behave normally in all cases.

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">This PR is still a Draft: some checks will be skipped.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Comment by iamgabrielma** [comment]
👋 the failing test is unrelated to our changes, it comes from flakiness on `test_intervalGranularity_for_dates_with_days_difference_from_1_to_28()` which has already been fixed https://github.com/woocommerce/woocommerce-ios/pull/12157. Just waiting for it to be merged into trunk so we can update this branch to pass CI

File: WooCommerce/Classes/ViewRelated/Orders/Order Creation/AddOrderComponentsSection/AddOrderComponentsSection.swift [hunk 1]
```diff
@@ -45,10 +45,13 @@ struct AddOrderComponentsSection: View {
...
             Group {
                 addShippingRow
+                    .frame(height: Constants.rowHeight)
 
                 addCouponRow
+                    .frame(height: Constants.rowHeight)
 
                 addGiftCardRow
+                    .frame(height: Constants.rowHeight)
             }
```
Comment (joshheald) [inline WooCommerce/Classes/ViewRelated/Orders/Order Creation/AddOrderComponentsSection/AddOrderComponentsSection.swift]:
```suggestion
            Group {
                addShippingRow

                addCouponRow

                addGiftCardRow
            }
            .frame(minHeight: Constants.rowHeight)
```

Comment (joshheald) [reply WooCommerce/Classes/ViewRelated/Orders/Order Creation/AddOrderComponentsSection/AddOrderComponentsSection.swift]: Groups can be used to apply the same modifier to multiple views.

`minHeight` seems more appropriate to allow the buttons to grow with larger dynamic type.

File: WooCommerce/Classes/ViewRelated/Orders/Order Creation/CustomAmounts/OrderCustomAmountsSection.swift [hunk 2]
```diff
@@ -29,6 +29,7 @@ struct OrderCustomAmountsSection: View {
                 .accessibilityIdentifier(Accessibility.addCustomAmountIdentifier)
                 .buttonStyle(PlusButtonStyle())
             }
+            .frame(height: Layout.rowHeight)
```
Comment (joshheald) [inline WooCommerce/Classes/ViewRelated/Orders/Order Creation/CustomAmounts/OrderCustomAmountsSection.swift]:
```suggestion
            .frame(minHeight: Layout.rowHeight)
```

File: WooCommerce/Classes/ViewRelated/Orders/Order Creation/CustomerNoteSection/CustomerNoteSection.swift [hunk 3]
```diff
@@ -73,12 +73,12 @@ private struct CustomerNoteSectionContent: View {
 
     private func createOrderNotesView() -> some View {
         Group {
-            Spacer(minLength: Layout.verticalHeadlineSpacing)
             Button(Localization.addNotes) {
                 showEditNotesView.toggle()
             }
             .buttonStyle(PlusButtonStyle())
-            .padding([.leading, .bottom, .trailing])
+            .frame(height: Layout.buttonHeight)
```
Comment (joshheald) [inline WooCommerce/Classes/ViewRelated/Orders/Order Creation/CustomerNoteSection/CustomerNoteSection.swift]:
```suggestion
            .frame(minHeight: Layout.buttonHeight)
```

File: WooCommerce/Classes/ViewRelated/Orders/Order Creation/CustomerSection/OrderCustomerSection.swift [hunk 4]
```diff
@@ -66,8 +66,8 @@ private struct OrderCustomerSectionContent: View {
             .renderedIf(viewModel.isDataAvailable)
 
             if !viewModel.isDataAvailable {
-                Spacer(minLength: Layout.verticalHeadlineSpacing)
                 createCustomerView
+                    .frame(height: Layout.buttonHeight)
```
Comment (joshheald) [inline WooCommerce/Classes/ViewRelated/Orders/Order Creation/CustomerSection/OrderCustomerSection.swift]:
```suggestion
                    .frame(minHeight: Layout.buttonHeight)
```

**Review by joshheald — APPROVED** [review]
Looks much better.

I added a divider that was missing at the top of the `Scan Products` line in side-by-side. Strange as it shows in your screenshot! 35b9100

Some suggestions to make the row height easier to apply and work better with dynamic type – I've applied them all in 01b83fc.

Ideally we would have this in some sort of style that got automatically applied to the rows, but I'm not sure how to do it off the top of my head, and it'll probably mean we don't fix/test other things well if we try

**Comment by iamgabrielma** [comment]
> I added a divider that was missing at the top of the Scan Products line in side-by-side. Strange as it shows in your screenshot! https://github.com/woocommerce/woocommerce-ios/commit/35b910045ccdd667f442379fea13356220758da8

Good catch, I believe I know where this comes from: I realized we were applying a double `Divider()` to the top. The difference was not much, but enough to notice the 1px vs 2px in certain screens, so I removed it in the last commit here: https://github.com/woocommerce/woocommerce-ios/pull/12175/commits/94a8f008117a6d137c05957a109c4c6a400499c8 without realizing the second one was being applied to modal-on-modal only. Thanks for addressing it!

File: WooCommerce/Classes/ViewRelated/Orders/Order Creation/AddOrderComponentsSection/AddOrderComponentsSection.swift (hunk 1 above)
Comment (iamgabrielma) [reply WooCommerce/Classes/ViewRelated/Orders/Order Creation/AddOrderComponentsSection/AddOrderComponentsSection.swift]: > Groups can be used to apply the same modifier to multiple views.

🤦 I knew this one, not sure why I didn't use it. Thanks for updating it!

> minHeight seems more appropriate to allow the buttons to grow with larger dynamic type.

That makes sense!

**Pull request merged**